newIDE/app/src/MainFrame/index.js

```diff
@@ -1534,6 +1566,21 @@ class MainFrame extends React.Component<Props, State> {
     });
   };
 
+  _onSetPreview = (editorTab: number) => {
+    const { currentProject } = this.state;
+    //Get name of scene on tab clicked
+    let name = editorTab.editorRef.props.layoutName;
```

Comment:
If you're looking at the props of your children, it's almost always because you're not asking your children to do enough (your React children! 😄).
In this case, you find yourself searching for the `layoutName`. Instead, make your children to give you the name of the scene when they call `togglePreviewOverride`.

Reply:
In other words, can you change this function so that instead of `editorTab`, you're having `layoutName`. Then think backward: what do I need to do when calling this function to pass the layoutName?
Which component has knowledge of the layout name and can pass it?

Reply:
SceneEditor have the LayoutName, but I'm confuse how pass the value from children(SceneEditor) to parent(Mainframe).
I've already do that in past?🧐